lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# filter_ias({ repo => 'goodies', dev_milestone => 'live'... })
+# Lookups combine as an AND operation.
+sub filter_ias {
+	my ($lookups) = @_;
+	my @results = @ia_container;
+	while (my ($by, $lookup) = each %$lookups) {
+		return () unless @results;
+		my @lookup = ref $lookup eq 'ARRAY' ? @$lookup : ($lookup);
+		@results = grep {
+			my $ia = $_;
+			any { _satisfy($ia, $by, $_) } @lookup;
+		} @results;
+	}
+	return \@results;
```

Comment:
`by_id` sort of sets the expectation that you will be able to key into the data structure.  We could potentially use `wantarray` to detect if the caller wants a list, otherwise send reduced version of `by_id` as a scalar.